src/visualizers/SeqColor.ts

```diff
@@ -0,0 +1,439 @@
+    private boxIsShow = false
+    private primeNum: bigint[] = []
+    private countPrime = 0
+    private firstDraw = true
+    private showLabel = false
+    private brightAdjust = 100
+
+    constructor() {
+        super()
+        // It is mandatory to initialize the `evaluator` property here,
+        // so just use a simple dummy formula until the user provides one.
+        this.evaluator = math.compile(this.formula)
+    }
+
+    initialize(sketch: p5, seq: SequenceInterface) {
+        super.initialize(sketch, seq)
+        this.sketch = sketch
+        this.seq = seq
+        this.currentIndex = seq.first
+        this.position = this.sketch.createVector(0, 0)
+        this.ready = true
+        this.boxSize = this.sketch.createVector(800, 90)
+        this.subG = this.sketch.createGraphics(this.boxSize.x, this.boxSize.y)
+        this.canvasSize = this.sketch.createVector(800, 800)
+        this.subL = this.sketch.createGraphics(
+            this.canvasSize.x,
+            this.canvasSize.y
+        )
+    }
+
+    checkParameters() {
+        // code currently re-used from SequenceFormula.ts
+        const status = super.checkParameters()
+
+        let parsetree = undefined
+        try {
+            parsetree = math.parse(this.params.formula.value)
+        } catch (err: unknown) {
+            status.isValid = false
+            status.errors.push(
+                'Could not parse formula: ' + this.params.formula.value
+            )
+            status.errors.push((err as Error).message)
+            return status
+        }
+        const othersymbs = parsetree.filter(
+            (node, path, parent) =>
+                node.type === 'SymbolNode'
+                && parent?.type !== 'FunctionNode'
+                && node.name !== 'n'
+                && node.name !== 'x'
+        )
+        if (othersymbs.length > 0) {
+            status.isValid = false
+            status.errors.push(
+                "Only 'n' and 'x' may occur as a free variable in formula.",
+                `Please remove '${(othersymbs[0] as math.SymbolNode).name}'`
+            )
+        }
+        this.evaluator = parsetree.compile()
+        return status
+    }
+
+    growthFunction(n: number, x: number) {
+        return this.evaluator.evaluate({n: n, x: x})
+    }
+
+    setup() {
+        this.sketch.background('black')
+        this.sketch.colorMode(this.sketch.HSB, 360, 100, 100)
+        this.sketch.frameRate(30)
+
+        this.firstDraw = true
+
+        // Set position of the circle
+        this.position = this.sketch.createVector(
+            this.canvasSize.x / this.n + 30,
+            this.canvasSize.x / this.n + 50
+        )
+
+        // Obtain all prime numbers from the sequence
+        for (let i = this.seq.first; i < this.n; i++) {
+            const checkCurrentPrime = this.seq.getElement(i)
+            if (
+                this.isPrime(i)
+                && !this.primeNum.includes(checkCurrentPrime)
```

Comment:
this is odd, because it takes the prime values of the sequence, not prime divisors?  is that what we want?  those are the only primes which contribute to the colouring of the sequence, as the code stands.

Reply:
fixed; take a look

Reply:
Yes, seems to take the primes that occur as factors, not just those that occur as elements of the sequence.